tldw_Server_API/app/core/AuthNZ/username_utils.py

```
@@ -0,0 +1,29 @@
+_USERNAME_PATTERN = re.compile(r"^[a-zA-Z0-9_-]+$")
+_RESERVED_USERNAMES = {"admin", "root", "system", "api", "null", "undefined"}
+
+
+class InvalidUsernameError(ValueError):
+    """Raised when a username fails normalization or validation."""
+
+
+def normalize_admin_username(raw: str) -> str:
+    candidate = (raw or "").strip().lower()
+    if not candidate:
+        raise InvalidUsernameError("Username is required")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)